src/insight/database/utils.py

```diff
@@ -76,16 +77,34 @@ def get_version_id(version: str, session: Session) -> int:
     Args:
         version (str): The name of the version.
         session (Session): The database session.
+
+    Returns:
+        int: The id of the version.
     """
-    db_version = get_object_from_db_by_name(version, session, model.Version)
-    if db_version is None:
-        with session:
-            db_version = model.Version(name=version)
-            session.add(db_version)
-            session.commit()
-    if not db_version.id:
-        raise ConnectionError(_database_fail_note)
-    return int(db_version.id)
+    try:
+        db_version = get_object_from_db_by_name(version, session, model.Version)
+        if db_version is None:
+            with session.begin_nested():
+                db_version = model.Version(name=version)
+                session.add(db_version)
+                session.commit()
+
+        if not db_version.id:
+            raise ConnectionError(_database_fail_note)
+
+        return int(db_version.id)
+
+    except IntegrityError as e:
```

Comment:
How do we know this is the error we'll get?

Reply:
Violating constraints raises integrity errors)

Reply:
Haha sure 😂 My question was more how you know it'll be an `IntegrityError` rather than some other type of Connection Error?

Reply:
I have another exception block under that to handle other errors:

```python
    try:
        ...
    except IntegrityError as e:
        session.rollback()
        ...
    except SQLAlchemyError as e:
        session.rollback()
```

Reply:
It's just we can recover from the `IntegrityError` because we can fetch the model again (breaking from the race cond), but other errors will fail fast